domains/pallets/auto-id/src/benchmarking.rs

```diff
@@ -0,0 +1,176 @@
+//! Benchmarking for `pallet-auto-id`.
+
+use super::*;
+use crate::Identifier;
+use frame_benchmarking::v2::*;
+use frame_system::RawOrigin;
+
+#[allow(clippy::multiple_bound_locations)]
+#[benchmarks(
+    where T: pallet_timestamp::Config<Moment = u64>
+)]
+mod benchmarks {
+    use super::*;
+
+    fn do_register_issuer<T: Config>() {
+        let account = account("issuer", 0, 0);
+        let issuer_register_params = RegisterAutoId::decode(
+            &mut include_bytes!("../res/benchmarks/issuer_register_auto_id_params").as_slice(),
+        )
+        .unwrap();
+        Pallet::<T>::register_auto_id(RawOrigin::Signed(account).into(), issuer_register_params)
+            .unwrap();
+    }
+
+    fn do_register_leaf<T: Config>() {
+        let account = account("leaf", 0, 0);
+        let leaf_register_params = RegisterAutoId::decode(
+            &mut include_bytes!("../res/benchmarks/leaf_register_auto_id_params").as_slice(),
+        )
+        .unwrap();
+        Pallet::<T>::register_auto_id(RawOrigin::Signed(account).into(), leaf_register_params)
+            .unwrap();
+    }
+
+    #[benchmark]
+    fn register_issuer_auto_id() {
+        let issuer_register_params = RegisterAutoId::decode(
+            &mut include_bytes!("../res/benchmarks/issuer_register_auto_id_params").as_slice(),
+        )
+        .unwrap();
+        let account = account("issuer", 0, 0);
+        pallet_timestamp::Pallet::<T>::set_timestamp(1_719_792_000_000u64);
```

Comment:
Not true actually. We use timestamp as `now()` since we are unable to generate certificates on the fly. Certificates comes with validity timestamps and our value of `now()` should be with the expiry time.

If we really want to get rid of this, then we should figure out a way to generate certificates on fly in `no_std` unfortunately.